### PR TITLE
Clippy cleanup

### PR DESCRIPTION
**Pull request by asomers**
This gets clippy to pass on mio.  I'll submit a separate PR for the trivial-pass-by-ref issues in public methods.

### CODE REVIEW

File: src/poll.rs [hunk 1]
```diff
@@ -1879,7 +1880,10 @@ impl RegistrationInner {
         // pointer is being operated on. The actual memory is guaranteed to be
         // visible the `poll: &Poll` ref passed as an argument to the function.
         let mut queue = self.readiness_queue.load(Relaxed);
-        let other: &*mut () = unsafe { mem::transmute(&poll.readiness_queue.inner) };
+        let other: &*mut () = unsafe {
+            &*(&poll.readiness_queue.inner as *const _ as *const *mut ())
+        };
+        //let other: &*mut () = unsafe { mem::transmute(&poll.readiness_queue.inner) };
```
[the remark this answers is not in the document]
Comment (asomers) [reply src/poll.rs]: This is the "transmute_ptr_to_ptr" lint.  The idea is that `mem::transmute` is so powerful that it's easy to use incorrectly.  The compiler does more checking when you cast pointers than when you use `mem::transmute`.  The comment, of course, is a mistake.  I'll remove it.

File: src/poll.rs [hunk 2]
```diff
@@ -2648,7 +2655,7 @@ impl ReadinessState {
     }
 
     #[inline]
-    fn is_queued(&self) -> bool {
+    fn is_queued(self) -> bool {
```
[the remark this answers is not in the document]
Comment (asomers) [reply src/poll.rs]: Yeah, but it's a private struct, so there's no harm in an API change.  This is the "trivially_copy_pass_by_ref" lint.  Basically, Clippy is saying that it's faster to pass `ReadinessState` by value than by reference.

[the remark this answers is not in the document]
Comment (tobz) [reply src/poll.rs]: Interesting.  The performance aspect makes sense.  I guess if the tests pass, it's fine.. but looking at this initially made me thing we'd be cloning/copying it all over the pass as every introspection passed it by value.

File: src/poll.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobz) [reply src/poll.rs]: Yeah, as a not core maintainer, your explanation makes sense to me but I can't help but think that the "safer" approach is also immeasurably harder to read.  Like, I almost have no idea what it's doing... but I also have no idea why you'd want a mutable reference to a unit value, so... :)

**Comment by carllerche** [comment]
As long as it doesn't touch public APIs, I'm fine with the idea behind this.

File: src/poll.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (asomers) [reply src/poll.rs]: Exactly: we will be copying all over the place.  But copying an 8-byte value is cheaper than creating an 8-byte pointer to that same value.

**Comment by asomers** [comment]
@carllerche after this PR merges, I'll create another one for the backwards-incompatible API changes.

**Comment by carllerche** [comment]
@asomers I don't know if I would do the breaking change PR yet... maybe wait until we know when 0.6 will realistically happen. I would hate for it to get bit rotted.

File: src/poll.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobz) [reply src/poll.rs]: Fair point!

**Pull request merged**